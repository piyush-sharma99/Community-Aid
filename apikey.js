/*
 *  ClassName: apikey.js
 *
 *  Date: 28/03/2021
 *
 * @author Piyush Sharma, X17342356
 *
 * @reference https://reactnative.dev/docs/activityindicator
 * @reference https://www.udemy.com/course/react-native-the-practical-guide/
 * @reference https://www.youtube.com/watch?v=8zT6CYu0iYQ
 *
 */

export const firebaseConfig = {
  apiKey: "AIzaSyDGliQfMfKpXUllG5ZPyClJL6C8VgFKXo8",
  authDomain: "community-aid.firebaseapp.com",
  databaseURL:
    "https://community-aid-default-rtdb.europe-west1.firebasedatabase.app",
  projectId: "community-aid",
  storageBucket: "community-aid.appspot.com",
  messagingSenderId: "529896701339",
  appId: "1:529896701339:web:1c30369d39f8654b01353e",
  measurementId: "G-PPD28P0PC6",
};
