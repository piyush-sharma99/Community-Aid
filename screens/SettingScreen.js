import React from 'react';
import { StyleSheet, Text, View, TextInput, Button } from 'react-native';
import SettingForm from '../components/SettingForm';

const SettingScreen = props => {
    return(
        <View style={styles.screen}>
            <SettingForm />
        </View>
    );
};

SettingScreen.navigationOptions = {
    headerTitle: 'Settings',
    headerStyle: {
    backgroundColor: '#2E86C1'
    },
    headerTintColor:"white"
    };

const styles = StyleSheet.create({
    screen: {
        justifyContent:"center",
        flex: 1,
        backgroundColor: '#2E86C1',
        paddingRight:60,
        paddingLeft:60
    }
});

export default SettingScreen;