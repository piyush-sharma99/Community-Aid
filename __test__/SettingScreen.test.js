import React from "react";
import SettingScreen from "../screens/SettingScreen";
import { firebaseConfig } from "../apikey";
import { cleanup } from "@testing-library/react-native";
import renderer from "react-test-renderer";
import * as firebase from "firebase";

firebase.initializeApp(firebaseConfig);

afterEach(cleanup);

test("Settings renders correctly", () => {
  const tree = renderer.create(<SettingScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
