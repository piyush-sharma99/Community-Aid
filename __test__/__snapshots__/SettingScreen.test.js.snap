// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollIntoView={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
    update={[Function]}
    viewIsInsideTabBar={false}
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "height": "100%",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignSelf": "stretch",
              "color": "#fff",
              "fontSize": 30,
              "justifyContent": "center",
              "marginTop": -100,
            }
          }
        >
          <Text
            style={
              Object {
                "borderColor": "#fb5b5a",
                "borderWidth": 3,
                "color": "white",
                "fontSize": 25,
                "fontWeight": "bold",
                "marginBottom": 30,
                "marginLeft": 40,
                "marginRight": 40,
                "marginTop": 130,
                "padding": 10,
                "textAlign": "center",
              }
            }
          >
            Edit account information below:
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fb5b5a",
                "borderColor": "white",
                "borderRadius": 25,
                "borderWidth": 3,
                "elevation": 10,
                "justifyContent": "center",
                "marginLeft": 20,
                "marginRight": 20,
                "marginTop": 20,
                "padding": 10,
                "shadowColor": "black",
                "shadowOffset": Object {
                  "height": 3,
                  "width": 3,
                },
                "shadowOpacity": 1,
                "shadowRadius": 10,
                "width": "90%",
              }
            }
          >
            <View
              accessible={true}
              focusable={false}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flexGrow": 1,
                  "flexShrink": 1,
                }
              }
            >
              <TextInput
                allowFontScaling={true}
                index={0}
                onChangeText={[Function]}
                placeholder="Change Name"
                placeholderTextColor="white"
                rejectResponderTermination={true}
                siblings={
                  Array [
                    "Component",
                    "Component",
                    "Component",
                    "Component",
                    "View",
                  ]
                }
                style={
                  Object {
                    "borderBottomWidth": 2,
                    "borderColor": "white",
                    "color": "#fff",
                    "fontSize": 25,
                    "height": 30,
                    "marginBottom": 30,
                    "marginTop": 30,
                  }
                }
                total={5}
                underlineColorAndroid="transparent"
              />
              <TextInput
                allowFontScaling={true}
                index={1}
                onChangeText={[Function]}
                placeholder="Change Number"
                placeholderTextColor="white"
                rejectResponderTermination={true}
                siblings={
                  Array [
                    "Component",
                    "Component",
                    "Component",
                    "Component",
                    "View",
                  ]
                }
                style={
                  Object {
                    "borderBottomWidth": 2,
                    "borderColor": "white",
                    "color": "#fff",
                    "fontSize": 25,
                    "height": 30,
                    "marginBottom": 30,
                  }
                }
                total={5}
                underlineColorAndroid="transparent"
              />
              <TextInput
                allowFontScaling={true}
                index={2}
                onChangeText={[Function]}
                placeholder="Change Email"
                placeholderTextColor="white"
                rejectResponderTermination={true}
                siblings={
                  Array [
                    "Component",
                    "Component",
                    "Component",
                    "Component",
                    "View",
                  ]
                }
                style={
                  Object {
                    "borderBottomWidth": 2,
                    "borderColor": "white",
                    "color": "#fff",
                    "fontSize": 25,
                    "height": 30,
                    "marginBottom": 30,
                  }
                }
                total={5}
                underlineColorAndroid="transparent"
              />
              <TextInput
                allowFontScaling={true}
                index={3}
                onChangeText={[Function]}
                placeholder="Change Password"
                placeholderTextColor="white"
                rejectResponderTermination={true}
                secureTextEntry={true}
                siblings={
                  Array [
                    "Component",
                    "Component",
                    "Component",
                    "Component",
                    "View",
                  ]
                }
                style={
                  Object {
                    "borderBottomWidth": 2,
                    "borderColor": "white",
                    "color": "#fff",
                    "fontSize": 25,
                    "height": 30,
                    "marginBottom": 30,
                  }
                }
                total={5}
                underlineColorAndroid="transparent"
              />
              <View
                index={4}
                siblings={
                  Array [
                    "Component",
                    "Component",
                    "Component",
                    "Component",
                    "View",
                  ]
                }
                total={5}
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#2E86C1",
                      "borderColor": "white",
                      "borderRadius": 15,
                      "borderWidth": 2,
                      "elevation": 10,
                      "height": 40,
                      "justifyContent": "center",
                      "marginBottom": 30,
                      "marginTop": 10,
                      "opacity": 1,
                      "shadowOpacity": 1,
                      "width": "100%",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "white",
                        "fontSize": 20,
                      }
                    }
                  >
                    Update Profile
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "alignSelf": "stretch",
                "borderBottomWidth": 2,
                "borderColor": "#fb5b5a",
                "color": "#fff",
                "fontSize": 30,
                "justifyContent": "center",
                "marginBottom": 20,
                "marginLeft": 10,
                "marginRight": 10,
                "marginTop": 20,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "marginLeft": 40,
                  "marginRight": 40,
                  "padding": 15,
                  "textAlign": "center",
                }
              }
            >
               Delete Account Below: 
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FDFEFE",
                "borderColor": "#fb5b5a",
                "borderRadius": 25,
                "borderWidth": 4,
                "elevation": 10,
                "height": 35,
                "justifyContent": "center",
                "marginBottom": 15,
                "marginLeft": 40,
                "marginTop": 15,
                "padding": 20,
                "width": 330,
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              onChangeText={[Function]}
              placeholder="Type 'Confirm' here then press delete..."
              placeholderTextColor="#003f5c"
              rejectResponderTermination={true}
              style={
                Object {
                  "color": "black",
                  "height": 50,
                }
              }
              underlineColorAndroid="transparent"
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fb5b5a",
                "borderColor": "white",
                "borderRadius": 15,
                "borderWidth": 2,
                "elevation": 10,
                "height": 40,
                "justifyContent": "center",
                "marginBottom": 10,
                "marginLeft": 60,
                "marginTop": 10,
                "opacity": 1,
                "shadowOpacity": 1,
                "width": "70%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 20,
                }
              }
            >
              Delete
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
